src/controllers/apiAuth.controller.ts

```diff
@@ -0,0 +1,62 @@
+import passport from "passport";
+import passportCustom from "passport-custom";
+import bcrypt from "bcrypt";
+const CustomStrategy = passportCustom.Strategy;
+import { pool } from "../config/database";
+
+async function hashPassword(password: string) {
+  const saltRounds = 10;
+  return await bcrypt.hash(password, saltRounds);
+}
+
+passport.use(
+  "custom-api-key",
+  new CustomStrategy(async (req, done) => {
+    try {
+      const phone_number = req.body.phone_number;
+      const password = req.body.password;
+
+      // Check if the user exists
+      const query = {
+        text: "SELECT * FROM celebration.user WHERE phone_number = $1",
+        values: [phone_number],
+      };
+      const result = await pool.query(query);
+
+      if (result.rows.length > 0) {
+        const user = result.rows[0];
+        const match = await bcrypt.compare(password, user.password);
+
+        if (match) {
+          // Passwords match, return the user info
+          return done(null, user);
+        } else {
+          // Passwords don't match
+          return done(null, { message: "Invalid Credentials" });
+        }
+      } else {
+        // User doesn't exist, create a new user with API key
+        const hashedPassword = await hashPassword(password);
+        const api_key = () => {
```

Comment:
This doesn't seem so secure. It is randomized to an extent, but not good enough for something like an API key.

Have you tried writing a script to determine how long generating an API key using this approach will take to hit a collision?

You would be surprised, don't you think 😄 ?

Reply:
I have already work on generating apikey but it's on celebrant branch

Reply:
Please close all these PRs and open a new one with all the changes.

Also, please host Birthmark, so that I can easily test the endpoints from the documentation.